### DISCUSSION TITLE
我非常喜欢的约定式路由在UMI4中找不到了，是不建议使用了吗？

### DISCUSSION THREAD

**Discussion by laeni** [umijs/umi#8287] [General]
约定式路由可能在很多地方无法方便地完成某些操作，但是它有一个非常大的优点，那就是直观。

说到这里，我就想起曾经遇到一个公司项目，项目是用Vue写的，而路由配置这里很奇葩，路由和组件层级关系对不上就算了，关键是路由配置是后端返回的。在这种情况下，虽然看似很"灵活"，但是如果想根据浏览器的路由路径去项目中找对应的组件基本是不可能的，因为必须要去后台看配置。而约定式路由就不存在这个问题，它就想传统的那样，能让新人一看就会。这也是我非常喜欢umi3和nextjs的原因之一。

然而在UMI4中，约定式路由没了，不知道是不建议使用（以后都没了）还是只是暂时没实现，希望只是暂时的吧。

**Comment by fz6m** [comment]
优先使用配置文件里明确指定的路由配置，如果不指定就是约定式路由，一直在的。

**Reply by laeni** [reply]
原来是把 children 换成了 &lt;Outlet /&gt;，升级之后不显示子路由，我以为被删了😃

**Reply by Pszz** [reply]
@fz6m @sorrycc 
请问这个约定方方式不支持了吗？

`pages/目录/_layout.tsx `

**Reply by fz6m** [reply]
> @fz6m @sorrycc 请问这个约定方方式不支持了吗？
> 
> `pages/目录/_layout.tsx `

不支持了，用 wrappers 替代吧。

 - https://umijs.org/docs/guides/routes#wrappers
